### PR TITLE
BX-4155 Cancel All 

### PR DESCRIPTION
**Pull request by kdonthi**

### CODE REVIEW

File: examples/grpcclient/main.go [hunk 1]
```diff
@@ -314,3 +316,71 @@ func callPostSettleGRPC(g *provider.GRPCClient, ownerAddr, ooAddr string) {
...
+	rand.Seed(time.Now().UnixNano())
+	clientOrderID1 := rand.Uint64()
+	clientOrderID2 := rand.Uint64()
+	opts := provider.PostOrderOpts{
+		ClientOrderID:     clientOrderID1,
+		OpenOrdersAddress: ooAddr,
+		SkipPreFlight:     true,
+	}
+
+	// Make 2 orders in ob
```
Comment (kdonthi) [inline examples/grpcclient/main.go]: Place

File: examples/grpcclient/main.go [hunk 2]
```diff
@@ -314,3 +316,71 @@ func callPostSettleGRPC(g *provider.GRPCClient, ownerAddr, ooAddr string) {
...
+		log.Fatal(err)
+	}
+	log.Infof("submitting place order #2, signature %s", sig)
+
+	time.Sleep(time.Minute)
+
+	// Check orders are there
+	orders, err := g.GetOpenOrders(ctx, marketAddr, owner)
+	found1 := false
+	found2 := false
```
Comment (kdonthi) [inline examples/grpcclient/main.go:368]: add a bit more spacing after 351 and 353

File: examples/httpclient/main.go [hunk 3]
```diff
@@ -229,3 +232,69 @@ func callPostSettleHTTP(ownerAddr, ooAddr string) {
...
+
+	rand.Seed(time.Now().UnixNano())
+	clientOrderID1 := rand.Uint64()
+	clientOrderID2 := rand.Uint64()
+	opts := provider.PostOrderOpts{
+		ClientOrderID:     clientOrderID1,
+		OpenOrdersAddress: ooAddr,
+		SkipPreFlight:     true,
+	}
+	// Make 2 orders in ob
```
Comment (kdonthi) [inline examples/httpclient/main.go]: Place + space

File: examples/grpcclient/main.go [hunk 4]
```diff
@@ -314,3 +322,76 @@ func callPostSettleGRPC(g *provider.GRPCClient, ownerAddr, ooAddr string) {
...
+		if order.ClientOrderID == fmt.Sprintf("%v", clientOrderID1) {
+			found1 = true
+			continue
+		}
+		if order.ClientOrderID == fmt.Sprintf("%v", clientOrderID2) {
+			found2 = true
+		}
+	}
+	if !(found1 && found2) {
+		log.Fatal("both orders not found in orderbook")
```
Comment (kdonthi) [inline examples/grpcclient/main.go]: get #

File: examples/wsclient/main.go [hunk 5]
```diff
@@ -366,3 +374,74 @@ func callPostSettleWS(w *provider.WSClient, ownerAddr, ooAddr string) {
...
+	for _, order := range orders.Orders {
+		if order.ClientOrderID == fmt.Sprintf("%v", clientOrderID1) {
+			found1 = true
+			continue
+		}
+		if order.ClientOrderID == fmt.Sprintf("%v", clientOrderID2) {
+			found2 = true
+		}
+	}
+
```
Comment (kdonthi) [inline examples/wsclient/main.go]: space

**Review by aspin — APPROVED** [review]

File: bxserum/provider/ws.go [hunk 6]
```diff
@@ -356,6 +356,43 @@ func (w *WSClient) SubmitCancelByClientOrderID(
 	return w.signAndSubmit(order.Transaction, skipPreFlight)
 }
 
+func (w *WSClient) PostCancelAll(market, owner string, openOrdersAddresses []string) (*pb.PostCancelAllResponse, error) {
+	request, err := w.jsonRPCRequest("PostCancelAll", &pb.PostCancelAllRequest{
```
Comment (aspin) [inline bxserum/provider/ws.go]: have you merged from develop? this look out of date

Comment (kdonthi) [reply bxserum/provider/ws.go]: Which part looks out of date? The `PostCancelAll` method?

Comment (aspin) [reply bxserum/provider/ws.go]: yeah, I deleted `jsonRPCRequest`
Screen Shot 2022-07-21 at 5 23 16 PM

**Review by aspin — APPROVED** [review]

**Pull request merged**